pkg/executables/cmk.go

```diff
@@ -0,0 +1,371 @@
+	}
+	logger.MarkPass("Connected to CloudStack server")
+	return nil
+}
+
+func (c *Cmk) execWithNameAndIdFilters(ctx context.Context, parameterValue string, genericArgs ...string) (stdout bytes.Buffer, err error) {
+	argsWithNameFilterArg := append(genericArgs, fmt.Sprintf("name=\"%s\"", parameterValue))
+	result, err := c.exec(ctx, argsWithNameFilterArg...)
+	if err != nil {
+		return result, fmt.Errorf("error getting resource info filtering by id %s: %v", parameterValue, err)
+	}
+	if result.Len() == 0 {
+		msg := fmt.Sprintf("No resources found with name %s. Trying again filtering by id instead", parameterValue)
+		argsWithIdFilterArg := append(genericArgs, fmt.Sprintf("id=\"%s\"", parameterValue))
+		logger.V(6).Info(msg)
+		result, err = c.exec(ctx, argsWithIdFilterArg...)
+		if err != nil {
+			return result, fmt.Errorf("error getting resource info filtering by id %s: %v", parameterValue, err)
+		}
+	}
+	return result, nil
+}
+
+func (c *Cmk) exec(ctx context.Context, args ...string) (stdout bytes.Buffer, err error) {
+	c.setupExecConfig()
+	envMap, err := c.getEnvMap()
+	if err != nil {
+		return bytes.Buffer{}, fmt.Errorf("failed get environment map: %v", err)
+	}
+	err = c.buildCmkConfigFile(envMap)
+	if err != nil {
+		return bytes.Buffer{}, fmt.Errorf("failed cmk validations: %v", err)
+	}
+	argsWithConfigFile := append([]string{"-c", c.execConfig.ConfigFile}, args...)
+
+	return c.executable.Execute(ctx, argsWithConfigFile...)
+}
+
+// TODO: Add support for passing in domain from Deployment Config Spec
+func (c *Cmk) buildCmkConfigFile(envMap map[string]string) (err error) {
+	decodedString, err := b64.StdEncoding.DecodeString(envMap[cloudStackb64EncodedSecretKey])
+	if err != nil {
+		return fmt.Errorf("failed to decode value for %s with base64: %v", cloudStackb64EncodedSecretKey, err)
+	}
+	cfg, err := ini.Load(decodedString)
+	if err != nil {
+		return fmt.Errorf("failed to extract values from %s with ini: %v", cloudStackb64EncodedSecretKey, err)
+	}
+	section, err := cfg.GetSection("Global")
+	if err != nil {
+		return fmt.Errorf("failed to extract section 'Global' from %s: %v", cloudStackb64EncodedSecretKey, err)
+	}
+	apiKey, err := section.GetKey("api-key")
+	if err != nil {
+		return fmt.Errorf("failed to extract value of 'api-key' from %s: %v", cloudStackb64EncodedSecretKey, err)
+	}
+	secretKey, err := section.GetKey("secret-key")
+	if err != nil {
+		return fmt.Errorf("failed to extract value of 'secret-key' from %s: %v", cloudStackb64EncodedSecretKey, err)
+	}
+	apiUrl, err := section.GetKey("api-url")
+	if err != nil {
+		return fmt.Errorf("failed to extract value of 'api-url' from %s: %v", cloudStackb64EncodedSecretKey, err)
+	}
+	cmkInsecure, err := strconv.ParseBool(envMap[cloudmonkeyInsecureKey])
+	if err != nil {
+		return fmt.Errorf("failed to parse boolean value from %s: %v", cloudmonkeyInsecureKey, err)
+	}
+	cmkVerifyCert := strconv.FormatBool(!cmkInsecure)
+	t := templater.New(c.writer)
```

Comment:
So this file would be written under the `generated` folder right?

Reply:
That's right!